libc/include/llvm-libc-types/pthread_condattr_t.h

```diff
@@ -0,0 +1,18 @@
+//===-- Definition of pthread_condattr_t type -----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifndef LLVM_LIBC_TYPES_PTHREAD_CONDATTR_T_H
+#define LLVM_LIBC_TYPES_PTHREAD_CONDATTR_T_H
+
+#include "clockid_t.h"
+
+typedef struct {
+  clockid_t clock;
+  int pshared;
+} pthread_condattr_t;
```

Comment:
FWIW, POSIX doesn't really say what this opaque type needs to look like.  It just needs to store a `clockid_t` (4 possible values, 2 bits of storage) and 1 bit for whether the condition variable is process shared or not.

Bionic packs this into a single long, and makes the fields opaque.

Do we use bitfields at all for our ABI?  LLVM has ADT/BitFields.h which is really nice; folks have requested I use that class in LLVM in the past rather than vanilla C bitfields. IDK

Reply:
`pthread_condattr_t` is also a "relic of pthreads" which was left behind for C11 threads. So "who cares" I'm tempted to say.

Reply:
For bitfields we usually use enums (e.g. https://github.com/llvm/llvm-project/blob/main/libc/src/stdio/printf_core/core_structs.h#L34)
As for `ADT/BitFields.h` we could potentially create a version in `__support` if it's necessary/useful.

For the public API, I don't have a strong opinion on whether it's opaque or not.

Reply:
For a publicly consumable header (anything under include/) we cannot use C++ only language features such as `enum class` or C23 enum with underlying type.

> if it's necessary/useful.

Looking at how complex ADT/BitFields.h, I don't feel that it's useful enough here to justify porting that to llvm-libc.

I don't think optimizing `pthread_condattr_t` for size is really all that important TBH.